API/Endpoints/ExerciseEndpoints.cs

```diff
@@ -102,9 +108,14 @@ public static WebApplication UseExerciseEndpoints(this WebApplication app)
             var result = await service.SubmitSolutionAsync(dto, exerciseId, userId);
             if (result.IsFailed)
             {
-                return TypedResults.BadRequest(result.Errors.Select(e => e.Message).ToArray());
+                return TypedResults.Problem(statusCode: 500, title: "An error occured", detail: result.Errors.First().Message);
```

Comment:
Considering how clean this way of returning message is, and how 'clumsy' it is at the other endpoint, we maybe should consider replicating the underlaying service functionality for the other endpoint.